src/io.jl

```diff
@@ -155,23 +155,19 @@ function _load(png_ptr, info_ptr; gamma::Union{Nothing,Float64}=nothing, expand_
     # Gamma correction is applied to a palette after `png_read_update_info` is called
     if read_as_paletted
         palette_length = Ref{Cint}()
-        # TODO: Figure out the lenght of paletted before calling png_get_PLTEs
         palette_buffer = Vector{RGB{N0f8}}(undef, PNG_MAX_PALETTE_LENGTH)
```

Comment:
For the removed comment, can't we use `png_get_palette_max(png_ptr, info_ptr)` documented in http://www.libpng.org/pub/png/libpng-manual.txt, and thus avoid using `PNG_MAX_PALETTE_LENGTH` ?

Reply:
I tried that and I can't get any other output than 0 (I did call it after `png_read_image`). I think I tried it before and gave up:). It is a good point, though, just don't know how to make it work. I'll keep the comment for now.

Reply:
I'll create an issue for this. If the approach you discovered (using transformations) is faster, it would be nice to get rid of this hack:)